### PR TITLE
atlas: remove

### LINKED ISSUE #49594: atlas fails to build
`atlas` has been broken in master since 2018-10-18:

https://hydra.nixos.org/build/82965121

The last successful build was 2018-10-13:

https://hydra.nixos.org/build/82713877

Here's the log tail of the failed build:

https://hydra.nixos.org/build/82965121/nixlog/1/tail

### PR DESCRIPTION
**Pull request by matthewbauer**
atlas is broken and can apparently be removed.

Fixes #49594

### CODE REVIEW

**Comment by markuskowa** [comment]
`blas` and `liblapack` may be replaced by `openblasCompat` here but this step requires further testing.
CC @symphorien @veprbl

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21203,13 +21191,7 @@ with pkgs;
 
   libhomfly = callPackage ../development/libraries/science/math/libhomfly { };
 
-  # We have essentially 4 permutations of liblapack: version 3.4.1 or 3.5.0,
-  # and with or without atlas as a dependency. The default `liblapack` is 3.4.1
-  # with atlas. Atlas, when built with liblapack as a dependency, uses 3.5.0
-  # without atlas. Etc.
   liblapack = callPackage ../development/libraries/science/math/liblapack {};
-  liblapackWithoutAtlas = liblapackWithAtlas.override { atlas = null; };
-  liblapackWithAtlas = liblapack;
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix]: `liblapackWithoutAtlas` would need to go to `aliases.nix`

**Comment by matthewbauer** [comment]
@grahamcofborg build petsc openmodelica shogun giac openblas

**Comment by markuskowa** [comment]
Using openblas instead of blas seems to work here (I can compile locally).
```
--- a/pkgs/applications/science/machine-learning/shogun/default.nix
+++ b/pkgs/applications/science/machine-learning/shogun/default.nix
@@ -2,7 +2,7 @@
 # data, compression
 , bzip2, curl, hdf5, json_c, lzma, lzo, protobuf, snappy
 # maths
-, blas, eigen, nlopt, lp_solve, colpack
+, openblasCompat, eigen, nlopt, lp_solve, colpack
 # libraries
 , libarchive, liblapack, libxml2
 # extra support
@@ -41,7 +41,7 @@ stdenv.mkDerivation rec {
   CCACHE_DIR=".ccache";
 
   buildInputs = with lib; [
-      blas bzip2 ccache cmake colpack curl ctags eigen hdf5 json_c lp_solve lzma lzo
+      openblasCompat bzip2 ccache cmake colpack curl ctags eigen hdf5 json_c lp_solve lzma lzo
       protobuf nlopt snappy swig (libarchive.dev) liblapack libxml2
     ]
     ++ optionals (pythonSupport) (with pythonPackages; [ python ply numpy ])
```

**Comment by markuskowa** [comment]
This turns out to be more work than though. 
- `liblapack` can be removed from shogun. It's not used anymore. Doesn't solve the aarch64 build failure though (?).
- `scalapack`  has never been built successfully on darwin (set `platforms.linux` should be fine here).

**Pull request merged**